### ISSUE TITLE
Not working on OSX

### ISSUE THREAD

**Issue by razeghi71** [adrianlyjak/obsidian-aloud-tts#26]
Hi, the plugin doesn't work for mac with this error in console:

```
app.js:1 Uncaught (in promise) Error: Request failed, status 429
    at new t (app.js:1:1977769)
    at qG (app.js:1:1977961)
    at app.js:1:1978638
    at app.js:1:237056
    at Object.next (app.js:1:237161)
    at a (app.js:1:235879)
```

I have the latest version (`0.2.0`) and obsidian version `1.5.12` 

Let me know if you need any additional info.

**Comment by adrianlyjak** [comment]
@razeghi71 that appears like you're getting a 429 request from openAI, which is a rate limit error code. 

Does the plugin ever work? Or do you immediately get the error. I would think either that 
A. something else is using the API key, exceeding your usage limits, 
B. or there's a bug in the plugin causing excessive requests 

Are you using the api key for anything else that might be using up your limits? If you navigate to the network tab, do you see many requests to openai? Are some of them successes? (200 error response, or are they all 429?)

**Comment by razeghi71** [comment]
I get this error when I click on the side button at the left, and there is no request actualy being sent at all, the network page shows nothing which returns 429, here are everything:

![Screenshot 2024-04-17 at 12 37 43](https://github.com/adrianlyjak/obsidian-aloud-tts/assets/1843462/bf09a28f-3b69-4c43-8169-049b55314d89)

**Comment by razeghi71** [comment]
Thanks for looking into this @adrianlyjak, yeah so it seems that this was related to the fact that OpenAI migrated from Personal API Keys to Project API Keys, so I've created a new project and there got an API Key and it worked!

**Comment by adrianlyjak** [comment]
@razeghi71 glad to hear you figured it out! I'll keep this open. It looks like I created some regressions in the error reporting with the latest release

**Comment by adrianlyjak** [comment]
Error messaging issues fixed in https://github.com/adrianlyjak/obsidian-aloud-tts/pull/30, released in 0.2.1

**Comment by adrianlyjak** [comment]
Error messaging issues fixed in https://github.com/adrianlyjak/obsidian-aloud-tts/pull/30, released in 0.2.1
